glean-core/csharp/Glean/Glean.cs

```diff
@@ -311,5 +320,35 @@ private void InitializeCoreMetrics()
             string storedvalue = GleanInternalMetrics.architecture.TestGetValue();
             Console.WriteLine("InitializeCoreMetrics - has value {0} and that's {1}", hasValue, storedvalue);
         }
-    }
+
+        /// <summary>
+        /// Register a [PingType] in the registry associated with this [Glean] object.
+        /// </summary>
+        internal void RegisterPingType(PingTypeBase pingType)
+        {
+            if (IsInitialized())
+            {
+                LibGleanFFI.glean_register_ping_type(
+                    pingType.handle
+                );
+            }
+
+            // We need to keep track of pings, so they get re-registered after a reset.
+            // This state is kept across Glean resets, which should only ever happen in test mode.
+            // Or by the instrumentation tests (`connectedAndroidTest`), which relaunches the application activity,
+            // but not the whole process, meaning globals, such as the ping types, still exist from the old run.
+            // It's a set and keeping them around forever should not have much of an impact.
+
+            //pingTypeQueue.Add(pingType);
```

Comment:
That's part of the solution to the problem with the type not surviving the tests :)

Reply:
Yep.